### PR TITLE
[PLAT-1301]

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Prevent too many submissions of the password reset token.

## Changes

- add check against submit_reset_token_last_attempt to ensure 10 second cooldown time between posting a password token.

## QA Notes

Not user facing.

## Documentation

🐞  fix, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/PLAT-1301

### CODE REVIEW

File: framework/auth/views.py [hunk 1]
```diff
@@ -35,6 +36,21 @@
...
 
+
+def throttle_password_reset(user):
+    if not throttle_period_expired(user.submit_reset_token_last_attempt, settings.TIME_RESET_SUBMIT_PASSWORD_TOKEN):
+        last_attempt = user.submit_reset_token_last_attempt.replace(tzinfo=pytz.utc)
+        time_since_throttle = (timezone.now() - last_attempt).total_seconds()
+        wait_time = int(settings.TIME_RESET_SUBMIT_PASSWORD_TOKEN - time_since_throttle)
+        error_data = {
+            'message_short': 'Too Many Requests.',
+            'message_long': 'The you have entered an incorrect token, please wait {} seconds to retry.'.format(wait_time)
```
Comment (erinspace) [inline framework/auth/views.py]: Remove the "The" -- "You have entered an incorrect token, please wait {} seconds to retry."

File: framework/auth/views.py [hunk 2]
```diff
@@ -87,6 +103,8 @@ def reset_password_post(uid=None, token=None):
 
     # Check if request bears a valid pair of `uid` and `token`
```
Comment (erinspace) [inline framework/auth/views.py]: This comment no longer makes sense here, maybe move it down above the real check

File: framework/auth/views.py [hunk 3]
```diff
@@ -35,6 +36,21 @@
...
+        time_since_throttle = (timezone.now() - last_attempt).total_seconds()
+        wait_time = int(settings.TIME_RESET_SUBMIT_PASSWORD_TOKEN - time_since_throttle)
+        error_data = {
+            'message_short': 'Too Many Requests.',
+            'message_long': 'The you have entered an incorrect token, please wait {} seconds to retry.'.format(wait_time)
+        }
+        raise HTTPError(429, data=error_data)
+    else:
+        user.submit_reset_token_last_attempt = timezone.now()
+        user.save()
```
Comment (erinspace) [inline framework/auth/views.py]: The user object is already being saved further down in `reset_password_post` when `verification_key_v2` is set, wonder if there's a way to potentially avoid a double user save in this method.

Comment (Johnetordoff) [reply framework/auth/views.py]: I tried putting this in an atomic transaction and a few other things and this is the best option I can see.

**Comment by Johnetordoff** [comment]
I didn't use `change_password_last_attempt` because I thought that someone could potentially get throttled by resetting their password and then quickly changing, but mostly I created a new field just to be more explicit.

**Pull request closed without merging**